### PR TITLE
Maze: don't serve animated gifs as pngs.

### PR DESCRIPTION
**Pull request by breville**
PvZ maze had two .png files that were actually animated .gif files, and IE11 refused to show them.  Angry Birds maze had one .png file in a similar state.  This change allows these levels to specify a .gif file for these cases.

It was a bit confusing that we used obstacle.png for the idle animation and obstacle.gif for the active animation, so I renamed obstacle.png to obstacleIdle.gif, etc., for this special-casing.

### CODE REVIEW

**Comment by breville** [comment]
@Bjvanminnen what do you think of this?

File: apps/src/maze/maze.js [hunk 1]
```diff
@@ -228,12 +228,12 @@ function drawMap () {
   pegmanFadeoutAnimation.setAttribute('begin', 'indefinite');
   pegmanIcon.appendChild(pegmanFadeoutAnimation);
 
-  if (Maze.finish_ && skin.goal) {
+  if (Maze.finish_ && (skin.goalIdle || skin.goal)) {
```
Comment (Bjvanminnen) [inline apps/src/maze/maze.js]: Let's just use one. It looks like goalIdle is the right thing to do. I looked at the other skins, and most don't actually have a goal.png. The one exception I found was scrat. Let's just have Scrat's config define goalIdle: 'goal.png'

Comment (Bjvanminnen) [inline apps/src/maze/maze.js]: Could also just use skin.goal, but set goal on the skin to be the .gif for those skins.

File: apps/src/maze/maze.js [hunk 2]
```diff
@@ -259,7 +259,7 @@ function drawMap () {
         obsIcon.setAttribute('height', Maze.MARKER_HEIGHT * skin.obstacleScale);
         obsIcon.setAttribute('width', Maze.MARKER_WIDTH * skin.obstacleScale);
         obsIcon.setAttributeNS(
-          'http://www.w3.org/1999/xlink', 'xlink:href', skin.obstacle);
+          'http://www.w3.org/1999/xlink', 'xlink:href', skin.obstacleIdle || skin.obstacle);
```
Comment (Bjvanminnen) [inline apps/src/maze/maze.js]: Again would prefer to stick to one, and make sure it gets set appropriately on the skin.

**Comment by breville** [comment]
thanks @Bjvanminnen .. updated per your feedback.  

(I also prefer using goalIdle/obstacleIdle explicitly to avoid inheritance override confusion.)

**Comment by Bjvanminnen** [comment]
:+1:

**Pull request merged**